### PR TITLE
[UEPR-17] Migrate webpack to version 5

### PR DESCRIPTION
**Pull request by MiroslavDionisiev**
### Jira Ticket
[UEPR-17](https://scratchfoundation.atlassian.net/browse/UEPR-17)

### Proposed Changes

Updated webpack to version 5 and other related dependencies. Used `scratch-webpack-configuration` to standardize the format of the webpack configuration between different application. Updated the version of `scratch-vm`.

### Test Coverage

- Build works, tests continue to pass
- Linked with scratch-vm, tests continue to pass
- Linked with scratch-gui, tests continue to pass

[UEPR-17]: https://scratchfoundation.atlassian.net/browse/UEPR-17?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -1,86 +1,76 @@
-const CopyWebpackPlugin = require('copy-webpack-plugin');
 const path = require('path');
-const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const webpack = require('webpack');
+
+const TerserPlugin = require('terser-webpack-plugin');
```
Comment (cwillisf) [inline webpack.config.js]: Can we skip Terser here and rely on webpack's `minimize` flag?

**Review by cwillisf — APPROVED** [review]
I left a question, but if the answer is "we need Terser for now" then I think this is fine as is. Thanks!

File: webpack.config.js (hunk 1 above)
Comment (MiroslavDionisiev) [reply webpack.config.js]: @cwillisf According to the webpack documentation for minimize:
> Tell webpack to minimize the bundle using the [TerserPlugin](https://webpack.js.org/plugins/terser-webpack-plugin/) or the plugin(s) specified in [optimization.minimizer](https://webpack.js.org/configuration/optimization/#optimizationminimizer).

In `scratch-webpack-configuration` we set `minimize` to true if the environment is production. So i guess we can remove the `minimizer` from the configuration here ans depend on what is set in the template.

Comment (cwillisf) [reply webpack.config.js]: That sounds good to me

**Pull request merged**